babel/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -505,13 +507,56 @@ SqlCreate SqlCreateTable(Span s, boolean replace) :
     |
         { query = null; }
     )
+    (
+       index = SqlCreateTableIndex(s)
+       {
+           if (index instanceof SqlPrimaryIndex) {
+               primaryIndex = (SqlPrimaryIndex) index;
+           }
+           //code for non-primary index support can be added here
+       }
+    )*
     onCommitType = OnCommitTypeOpt()
     {
         return new SqlCreateTable(s.end(this), replace, setType, volatility, ifNotExists, id,
-            tableAttributes, columnList, query, withData, onCommitType);
+            tableAttributes, columnList, query, withData, primaryIndex, onCommitType);
     }
 }
 
+/**
+Parses an index declaration.
+Currently only supports PRIMARY INDEX statements, but can be extended to support non-primary indices.
+*/
+SqlIndex SqlCreateTableIndex(Span s) :
+{
+   List<SqlNode> columns;
+   SqlIdentifier name = null;
+   boolean isUnique = false;
+}
+{
+   (
+       <NO> <PRIMARY> <INDEX>
+       {
+           return new SqlPrimaryIndex(s.end(this), /*columns=*/ null, /*name=*/ null, /*isUnique=*/ false, /*explicitNoPrimaryIndex=*/ true);
+       }
+   |
+       [
+           <UNIQUE> { isUnique = true; }
+       ]
+       <PRIMARY> <INDEX>
+       [
+           name = SimpleIdentifier()
+       ]
+       <LPAREN>
+       columns = SelectList()
```

Comment:
Is SelectList() the right parsing function to call here? According to its doc it: "Parses the select list of a SELECT statement."

Reply:
Maybe this entire section (including the left and right parens) could be replaced with
ParenthesizedSimpleIdentifierList?

Reply:
Thanks, I was looking for something like that and SelectList() seemed close enough, but I will keep this one in mind now.